modules/apps/27-interchain-accounts/module.go

```diff
@@ -100,22 +116,44 @@ func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sd
 	return nil
 }
 
-// RegisterServices registers a GRPC query service to respond to the
-// module-specific GRPC queries.
+// RegisterServices registers module services
 func (am AppModule) RegisterServices(cfg module.Configurator) {
-	types.RegisterQueryServer(cfg.QueryServer(), am.keeper)
 }
 
+// InitGenesis performs genesis initialization for the interchain accounts module.
+// It returns no validator updates.
 func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) []abci.ValidatorUpdate {
 	var genesisState types.GenesisState
 	cdc.MustUnmarshalJSON(data, &genesisState)
 
-	InitGenesis(ctx, am.keeper, genesisState)
+	if am.controllerKeeper != nil {
+		controllerkeeper.InitGenesis(ctx, *am.controllerKeeper, *genesisState.ControllerGenesisState)
+	}
+
+	if am.hostKeeper != nil {
+		hostkeeper.InitGenesis(ctx, *am.hostKeeper, *genesisState.HostGenesisState)
+	}
+
 	return []abci.ValidatorUpdate{}
 }
 
+// ExportGenesis returns the exported genesis state as raw bytes for the interchain accounts module
 func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
-	gs := ExportGenesis(ctx, am.keeper)
+	var (
+		controllerGenesisState *types.ControllerGenesisState
+		hostGenesisState       *types.HostGenesisState
```

Comment:
Ah yes! You're 100% right here. In fact, I think this could cause some problems if it marshalled using `nil` for one or the other and a chain attempted to restart with a genesis file provided. In that scenario the `DefaultGenesis` wouldn't be loaded upon restart, right?

Reply:
correct!